nestedtensor/csrc/nested_tensor_impl.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <nestedtensor/csrc/nested_tensor.h>
+
+namespace at {
+
+constexpr auto NestedTensorKey = DispatchKey::PrivateUse1_PreAutograd;
```

Comment:
Can you, please, add a comment explaining this? How and why it works here now and what is the plan for the future

Reply:
This is using one of the experimental dispatch keys, which would presumably be replaced by a fixed dispatch key in the future. Beyond that, a full explanation of the registration mechanism is to be found elsewhere.